pandas/tests/io/test_html.py

```diff
@@ -1286,3 +1286,57 @@ def test_parse_path_object(self, datapath):
         df1 = self.read_html(file_path_string)[0]
         df2 = self.read_html(file_path)[0]
         tm.assert_frame_equal(df1, df2)
+
+    def test_extract_hrefs(self):
```

Comment:
test is unnecssarily long. cn be minimised to just the use cases with comparisons,

e.g. 

- cell with an <a> tag and an hhtp link
- cell with an <a> tag and another kind of link e.g. ftp.
- cell without an <a> tag.

Could also include a header with an <a> tag to show it has no effect on header cells

Reply:
Done.